sdcm/nemesis.py

```diff
@@ -1817,6 +1812,31 @@ def disrupt_repair_streaming_err(self):
         self._set_current_disruption('RepairStreamingErr')
         self.break_streaming_task_and_rebuild(task='repair')
 
+    def disrupt_grow_shrink_cluster(self):
+        add_nodes_number = self.tester.params.get('add_node_cnt')
+
+        self._set_current_disruption("GrowCluster")
+        self.log.info("Start grow cluster on %s nodes", add_nodes_number)
+        for _ in range(add_nodes_number):
+            self.metrics_srv.event_start('add_node')
+            added_node = self._add_and_init_new_cluster_node()
+            added_node.running_nemesis = None
+            self.metrics_srv.event_stop('add_node')
+            self.log.info("New node added %s", added_node.name)
+        self.log.info("Finish cluster grow")
+
+        time.sleep(self.interval)
+
+        self._set_current_disruption("ShrinkCLuster")
```

Comment:
need to make sure that add node succeeded before deleting the node

Reply:
@bentsi if node wasn't added, then _add_and_init_new_cluster_node method will raise an exception, and whole nemesis will failed.
is it not enough? you think it is needed additional assertion?

Reply:
You are right, but what can happen, due to @retrying decorator that there will be more nodes than actually wanted...

```python
    @retrying(n=3, sleep_time=60, allowed_exceptions=(NodeSetupFailed, NodeSetupTimeout))
    def _add_and_init_new_cluster_node(self, old_node_ip=None, timeout=HOUR_IN_SEC * 6):
        """When old_node_private_ip is not None replacement node procedure is initiated"""
        self.log.info("Adding new node to cluster...")
```
but let's keep it as is.
@roydahan I remember that you insisted on the retry, what was the reason for that?